lib/matplotlib/testing/decorators.py

```diff
@@ -146,11 +148,8 @@ def setup_class(cls):
             mpl.rcParams.clear()
             mpl.rcParams.update(cls._initial_settings)
             raise
-        # Because the setup of a CleanupTest might involve
-        # modifying a few rcparams, this setup should come
-        # last prior to running the image test.
-        CleanupTest.setup_class()
         cls.original_settings = cls._initial_settings
+        matplotlib.tests.set_font_settings_for_testing()
```

Comment:
I like this solution to this problem. I got so tripped up by this when I was working on the classic mode PR.